File: codeeval/easy/167readmore/solution.rb
#!/usr/bin/env ruby

File.foreach(ARGV.first) do |line|
  line = line.strip()
  if line.size <= 55
    puts line
  else
    line = line[0,40]
    lsi = line.rindex(" ")
    line = line[0,lsi] if lsi
    puts "#{line}... <Read More>"
  end
end

File: CodeEval/easy/167readmore/solution.py
#!/usr/bin/env python

from sys import argv

argv = [None, "/Users/hlissner/Desktop/data"]

for line in open(argv[1]):
    line = line.strip()
    if len(line) > 55:
        line = line[0:40]
        lsi = line.rfind(" ")
        if lsi != -1:
            line = line[0:lsi+1]
        line = line.strip()
        line += "... <Read More>"
    print(line)
